### PR TITLE
Hints iteration refactoring

### LINKED ISSUE #645: Refactor `rewrite_steps.py`
# Intro

Right now, this file is preparing `RewriteStepExpression(pre_config, post_config, axiom, substitutions)` that are fed into the proof generator by zipping through the hint stream.

As `post_config` is never used, and we will have to remove zipping because of simplification events, make the following changes.

# Tasks

- [ ] Remove `conf_after` in `RewriteStepExpression` and all the associated methods. (This will actually be the next `conf_before` for the next rewrite, so this information is duplicate.)
- [ ] Remove the zipping from `get_proof_hints`, this is only used for `post_config` which is removed in step 1. Convert it into a normal forloop that goes through hint after another.
- [ ] Edit all the associated code and tests so that CI passes.

### PR DESCRIPTION
**Pull request by IljaZakharov**
Closes #645

Implements requested changes plus a couple of new ones:

- Fix the loop. But we keep processing patterns to fetch initial configurations for hints.
- Remove the post-configuration
- Remove the SIMPLE test. It contains hook events and functional events. We can't process them for now. It worked before due to an adhoc filtering that required the rewrite event followed by a configuration.

### CODE REVIEW

File: generation/src/tests/integration/rewrite/test_llvm_proof_generator.py [hunk 1]
```diff
@@ -108,24 +107,21 @@ def test_proof_trace_double_rewrite() -> None:
     hints_file = Path(HINTS_DIR + '/double-rewrite/foo-a.double-rewrite.hints')
     kompiled_dir = Path(KOMPILED_DIR + '/double-rewrite-kompiled/')
 
-    iterator = generate_proof_trace(k_file, hints_file, kompiled_dir)
+    initial_config, iterator = generate_proof_trace(k_file, hints_file, kompiled_dir)
+    assert initial_config
     assert iterator
```
Comment (nishantjr) [inline generation/src/tests/integration/rewrite/test_llvm_proof_generator.py:109]: I think these assertions are unnecessary because of typechecking?

**Review by nishantjr — APPROVED** [review]
Looks good.

File: generation/src/proof_generation/k/kore_convertion/rewrite_steps.py [hunk 2]
```diff
@@ -67,28 +60,37 @@ def substitutions(self) -> dict[int, Pattern]:
 
 
 def get_proof_hints(
-    llvm_proof_hint: LLVMRewriteTrace,
+    llvm_proof_hints: LLVMRewriteTrace,
```
[the remark this answers is not in the document]
Comment (IljaZakharov) [reply generation/src/proof_generation/k/kore_convertion/rewrite_steps.py:55]: Fixed

File: generation/src/proof_generation/k/kore_convertion/rewrite_steps.py [hunk 3]
```diff
@@ -67,28 +60,37 @@ def substitutions(self) -> dict[int, Pattern]:
...
+) -> Iterator[RewriteStepExpression]:
+    """
+    Emits proof hints corresponding to the given LLVM rewrite trace.
+    Note that no hints will be generated if the trace is empty.
+    """
+    # TODO: process function/hook/rule events in llvm_proof_hint.pre_trace
+    current_config = initial_config
+    for event in llvm_proof_hints.trace:
+        # TODO: process function/hook events in llvm_proof_hint.strace
+        if isinstance(event, LLVMRuleEvent):
```
[the remark this answers is not in the document]
Comment (IljaZakharov) [reply generation/src/proof_generation/k/kore_convertion/rewrite_steps.py:80]: Fixed

File: generation/src/tests/integration/rewrite/test_llvm_proof_generator.py (hunk 1 above)
Comment (IljaZakharov) [reply generation/src/tests/integration/rewrite/test_llvm_proof_generator.py:109]: Removed.

**Pull request merged**